test/finish_dialog_test.rb

```diff
@@ -0,0 +1,80 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2015 SUSE LLC, All Rights Reserved.
+#
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact SUSE LLC.
+#
+# To contact SUSE about this file by physical or electronic mail, you may find
+# current contact information at www.suse.com.
+# ------------------------------------------------------------------------------
+
+require_relative "test_helper"
+
+describe Migration::FinishDialog do
+  describe ".run" do
+    it "displays the finish message" do
+      # check the displayed content
+      expect(Yast::Wizard).to receive(:SetContents) do |_title, content, _help, _back, _next|
+        term = content.nested_find do |t|
+          t.respond_to?(:value) && t.value == :RichText &&
+            t.params[2].match(/Congratulations!/)
+        end
+
+        expect(term).to_not eq(nil)
```

Comment:
"Declaring" `term`inside the block and then checking its value afterwards looks weird. It could even break in future versions of Ruby (and readers from other languages will get shocked :wink:)